### DISCUSSION TITLE
Make not identifying target for core.

### DISCUSSION THREAD

**Discussion by bowensmachinesense** [OpenMathLib/OpenBLAS#4790] [Q&A]
I encountered a problem with make not identifying the core to build from.  I'm building on a Pentium 4 and the core type is PRESCOTT.  I'm using the current source code.

When I do a make I get:

Makefile:185: *** OpenBLAS: Detecting CPU failed. Please set TARGET explicitly, e.g. make TARGET=your_cpu_target. Please read README for the detail..

The build does work with:
make TARGET=PRESCOTT

After some digging I found in the file cpuid_x86.c the function get_corytype() was returning CORE_UNKNOWN. It seems as if the nesting is wrong on the switch statement and some things are nested one level to deep.  My eyes are not what they were so I could be mistaken.

In that file if you go to about line 2395 you will see.

```
      case 15:
	if (model <= 0x2) return CORE_NORTHWOOD;
	else return CORE_PRESCOTT;
      }
    }
  }

```
I changed it just for my testing to..

```
     }
      case 15:
	if (model <= 0x2) return CORE_NORTHWOOD;
	else return CORE_PRESCOTT;
      }
  }

```
That fixed the issue for me so that make works without specifying the target.  I am not sure exactly where the switch nesting got off at, but you can see I just took it up one nesting level so that the 'case 15' now operates on the 'switch (family)' level.

I may be off base here but I just wanted to provide the information.

Thanks
Brad

**Comment by martin-frbg** [comment]
Indeed, this appears to have been broken by me three years ago (in #3442 for version 0.3.19) when adding entries for Alder Lake. Guess family 15 cpus are becoming too rare now for this to have come up earlier (or nobody bothered to report this). Thank you !
